src/Main.elm

```diff
@@ -1,22 +1,114 @@
-module Main exposing (Model, Msg(..), book, init, main, update, view)
+module Main exposing (Model, Msg(..), init, main, update, view)
 
 import BookData exposing (Book, allBooks)
 import Browser
-import Html exposing (Html, div, h1, img, li, text, ul)
+import Html exposing (Html, button, div, img, table, td, text, th, thead, tr)
 import Html.Attributes exposing (alt, class, src)
+import Html.Events exposing (onClick)
+import List exposing (filter, map, sum)
+import List.Extra as ListX exposing (find, setIf)
 
 
 
 ---- MODEL ----
 
 
 type alias Model =
-    { books : List Book }
+    { books : List Book, cart : Cart }
+
+
+type alias CartItem =
+    { id : BookID, quantity : Int }
+
+
+type alias Cart =
+    List CartItem
+
+
+type alias BookID =
+    Int
+
+
+removeFromCart : BookID -> Cart -> Cart
+removeFromCart bookID cart =
+    List.filter (\cartItem -> cartItem.id /= bookID) cart
+
+
+addToCart : BookID -> Cart -> Cart
+addToCart bookID cart =
+    let
+        mCartItem =
+            ListX.find (\cartItem -> cartItem.id == bookID) cart
```

Comment:
Since you use this several times, you could extract it into a helper FN `findItemInCart`, not a biggie though.